packages/bbot/data_stream/asm_intel/elasticsearch/ingest_pipeline/default.yml

```
@@ -3,156 +3,119 @@ description: Pipeline for ingesting BBOT ndjson files.
+    - UNIX
+    - ISO8601
+    ignore_failure: false
+
+####################################################
+# Ensure BBOT v1.x & v2.x SCAN event compatibility #
+####################################################
 - rename:
-    field: bbot.resolved_hosts
-    target_field: related.hosts
+    description: "Avoid type conflict between BBOT v1.x (string/keyword) and v2.x (object)"
+    if: ctx.bbot?.data?.SCAN instanceof Map
+    field: bbot.data.SCAN
+    target_field: bbot.data.SCAN_CONFIG
     ignore_missing: true
-- append:
-    field: host.ip
-    value: '{{{bbot.data.IP_ADDRESS}}}'
-    if: ctx.bbot?.data?.IP_ADDRESS != null
-- append:
-    field: url.full 
-    value: '{{{bbot.data.URL}}}'
-    if: ctx.bbot?.data?.URL != null
-    allow_duplicates: false
-- append:
-    field: url.full
-    value: '{{{bbot.data.FINDING.url}}}'
-    if: ctx.bbot?.data?.FINDING?.url != null
-    allow_duplicates: false
-- append:
-    field: url.full
-    value: '{{{bbot.data.SOCIAL.url}}}'
-    if: ctx.bbot?.data?.SOCIAL?.url != null
-    allow_duplicates: false
-- append:
-    field: url.full
-    value: '{{{bbot.data.STORAGE_BUCKET.url}}}'
-    if: ctx.bbot?.data?.STORAGE_BUCKET?.url != null
-    allow_duplicates: false
-- append:
-    field: url.full
-    value: '{{{bbot.data.TECHNOLOGY.url}}}'
-    if: ctx.bbot?.data?.TECHNOLOGY?.url != null
-    allow_duplicates: false
-- append:
-    field: url.full
-    value: '{{{bbot.data.VULNERABILITY.url}}}'
-    if: ctx.bbot?.data?.VULNERABILITY?.url != null
-    allow_duplicates: false
-- append:
-    field: url.full
-    value: '{{{bbot.data.WAF.url}}}'
-    if: ctx.bbot?.data?.WAF?.url != null
-    allow_duplicates: false
-- append:
-    field: url.full
-    value: '{{{bbot.data.WEBSCREENSHOT.url}}}'
-    if: ctx.bbot?.data?.WEBSCREENSHOT?.url != null
-    allow_duplicates: false
-- append:
-    field: url.domain 
-    value: '{{{bbot.data.FINDING.host}}}'
-    if: ctx.bbot?.data?.FINDING?.host != null
-    allow_duplicates: false
-- append:
-    field: url.domain 
-    value: '{{{bbot.data.PROTOCOL.host}}}'
-    if: ctx.bbot?.data?.PROTOCOL?.host != null
-    allow_duplicates: false
-- append:
-    field: url.domain 
-    value: '{{{bbot.data.TECHNOLOGY.host}}}'
-    if: ctx.bbot?.data?.TECHNOLOGY?.host != null
-    allow_duplicates: false
-- append:
-    field: url.domain 
-    value: '{{{bbot.data.VULNERABILITY.host}}}'
-    if: ctx.bbot?.data?.VULNERABILITY?.host != null
-    allow_duplicates: false
-- append:
-    field: url.domain 
-    value: '{{{bbot.data.WAF.host}}}'
-    if: ctx.bbot?.data?.WAF?.host != null
-    allow_duplicates: false
-- append:
-    field: url.domain 
-    value: '{{{host.name}}}'
-    if: ctx.host?.name != null
-    allow_duplicates: false
+
+- set:
+    description: "Set bbot.data.SCAN based on bbot.data.SCAN_CONFIG to emulate BBOT v1.x field type"
+    if: ctx.bbot?.data?.SCAN == null && ctx.bbot?.data?.SCAN_CONFIG?.name != null && ctx.bbot?.data?.SCAN_CONFIG?.id != null
```

```suggestion
    if: ctx.bbot?.data?.SCAN == null && ctx.bbot?.data?.SCAN_CONFIG?.name != null && ctx.bbot.data.SCAN_CONFIG?.id != null
```
Already known to be non-null due to the second expression.

---

Errr... not really, SCAN_CONFIG.name and SCAN_CONFIG.id may still be null even if SCAN was renamed to SCAN_CONFIG.

These fields only exist in a small sub-set of events, e.g. the event that documents that start of the scan. We only want to set this field if we're truly confident that we're looking at that type of event, as evidenced by having both SCAN_CONFIG.id and SCAN_CONFIG.name at that point in the pipeline.